services/sonar/sonar-base.js

```diff
@@ -43,13 +43,17 @@ module.exports = class SonarBase extends BaseJsonService {
     return { userKey: 'sonarqube_token' }
   }
 
-  async fetch({ sonarVersion, protocol, host, component, metricName }) {
-    let qs, url, schema
+  async fetch({ sonarVersion, server, component, metricName }) {
+    if (!server) {
```

Comment:
Why do we need this check explicitly? Can we not use `optionalUrl.required()` when specifying the query parameter schema?

Reply:
Fixed, thanks!